app/src/main/java/org/p2p/wallet/moonpay/ui/BuySolanaPresenter.kt

```diff
@@ -160,6 +162,10 @@ class BuySolanaPresenter(
                 buyResultAnalytics = BuyAnalytics.BuyResult.ERROR
                 view?.showMessage(buyResult.message)
             }
+            is MoonpayBuyResult.MinimumAmountError -> {
+                // May by only in case of new buy
+                throw IllegalArgumentException("MinimumAmountError may be only in case of new buy screen")
```

Comment:
it's important to create own class Exceptions to track this errors in Sentry and Crashlytics:
<img width="936" alt="Screen Shot 2022-09-08 at 14 03 09" src="https://user-images.githubusercontent.com/100791035/189106364-18cacc90-9dc7-4af3-b6e0-be40c1ea14fc.png">

instead of abstract IllegalStateException we can now see in which domain places errors occur

Reply:
Actually that error is just a stub. It could be thrown only in case of using new Buy components on old Buy screen